storage_service/common/datatable_utils.py

```diff
@@ -0,0 +1,159 @@
+            "display_start": self._get_int_parameter(query_dict, "iDisplayStart"),
+            "display_length": self._get_int_parameter(query_dict, "iDisplayLength"),
+            # sorting information
+            "sorting_column": self.sorting_column(query_dict),
+            # parameter sent by the client that the server needs
+            # to return back in order to identify each single rendering
+            "echo": self._get_int_parameter(query_dict, "sEcho", -1),
+        }
+
+    def sort(self, queryset):
+        sorting_column = self.params["sorting_column"]
+        if not sorting_column or sorting_column.get("index") is None:
+            return queryset
+        # the columns represented by these indexes can be sorted
+        # through the queryset using the order_by method
+        ORDER_BY_MAPPING = {
+            0: "uuid",
+            1: "origin_pipeline__description",
+            3: "size",
+            5: "replicated_package__uuid",
+        }
+        # these columns instead can't be sorted directly with the
+        # queryset so the queryset needs to be converted into a list
+        # and then sorted using helper methods
+        SORT_KEY_HELPERS_MAPPING = {
+            2: self.sort_by_full_path_key,
+            4: self.sort_by_package_type_key,
+            6: self.sort_by_status_key,
+            7: self.sort_by_fixity_date_key,
+            8: self.sort_by_fixity_status_key,
+        }
+        sort_descending = sorting_column.get("direction") == "desc"
+        if sorting_column["index"] in ORDER_BY_MAPPING:
+            field = ORDER_BY_MAPPING[sorting_column["index"]]
+            if sort_descending:
+                field = "-{}".format(field)
+            return queryset.order_by(field)
+        elif sorting_column["index"] in SORT_KEY_HELPERS_MAPPING:
+            key = SORT_KEY_HELPERS_MAPPING[sorting_column["index"]]
+            queryset = sorted(list(queryset), key=key, reverse=sort_descending)
+        else:
+            return queryset
+
+    def get_packages(self, queryset):
+        result = self.sort(queryset)
+        display_start = self.params["display_start"]
+        display_length = self.params["display_length"]
+        if display_start and display_length:
+            return result[display_start : (display_start + display_length)]
+        else:
+            return result
+
+    def sort_by_full_path_key(self, package):
+        return package.full_path
+
+    def sort_by_package_type_key(self, package):
+        return package.get_package_type_display()
+
+    def sort_by_status_key(self, package):
+        return package.get_status_display()
+
+    def sort_by_fixity_date_key(self, package):
+        # avoid comparing datetimes with None
+        return package.latest_fixity_check_datetime or timezone.now()
+
+    def sort_by_fixity_status_key(self, package):
+        return package.latest_fixity_check_result
+
+
+def build_table(query_dict):
+    return DataTable(query_dict)
```

Comment:
I think you can just skip this function.

Reply:
:+1: